osquery/core/windows/wmi.h

```diff
@@ -173,8 +181,9 @@ class WmiRequest {
  private:
   Status status_;
   std::vector<WmiResultItem> results_;
-  IWbemLocator* locator_{nullptr};
-  IWbemServices* services_{nullptr};
-  IEnumWbemClassObject* enum_{nullptr};
+
+  std::unique_ptr<IWbemLocator, decltype(impl::wmiObjectDeleter)> locator_{nullptr, impl::wmiObjectDeleter};
+  std::unique_ptr<IEnumWbemClassObject, decltype(impl::wmiObjectDeleter)> enum_{nullptr, impl::wmiObjectDeleter};
+  std::shared_ptr<IWbemServices> services_{static_cast<IWbemServices*>(nullptr), impl::wmiObjectDeleter};
```

Comment:
Sorry, I didn't get it. Why `services_` is `shared_ptr`, not `unique_ptr`?

Reply:
It's there in preparation for a future PR which will add an identical shared services pointer to the WmiResultItem class, for implementing some new functionality there. For reference, you can see that future PR here at https://github.com/trailofbits/osquery-pr/pull/8.

Reply:
As far as it is not obvious why it should be shared from this diff, may I ask you to keep it `unique` in this diff and change later if required?